src/Microsoft.AspNet.Authentication.OpenIdConnect/OpenIdConnectAuthenticationHandler.cs

```diff
@@ -520,6 +545,25 @@ protected override AuthenticationTicket AuthenticateCore()
             }
         }
 
+        protected virtual async Task<TokenResponse> RedeemAuthorizationCode(string authorizationCode)
+        {
+            var tokenRequestParameters = new Dictionary<string, string>()
+            {
+                { "client_id", Options.ClientId },
+                { "redirect_uri", Options.RedirectUri },
+                { "client_secret", Options.ClientSecret },
+                { "code", authorizationCode },
+                { "grant_type", "authorization_code" },
+            };
+            var requestMessage = new HttpRequestMessage(HttpMethod.Post, _configuration.TokenEndpoint);
+            requestMessage.Content = new FormUrlEncodedContent(tokenRequestParameters);
```

Comment:
In the OAuth2 client handler, we only support flowing the client credentials via the request body. It would be probably useful to also support `client_secret_basic`, specially since we can easily know if it's supported using the provider configuration if it's available.